ssvsigner/client.go

```diff
@@ -0,0 +1,205 @@
+		},
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+type ClientOption func(*Client)
+
+func WithLogger(logger *zap.Logger) ClientOption {
+	return func(client *Client) {
+		client.logger = logger
+	}
+}
+
+func (c *Client) ListValidators(ctx context.Context) ([]phase0.BLSPubKey, error) {
+	var resp web3signer.ListKeysResponse
+	err := requests.
+		URL(c.baseURL).
+		Client(c.httpClient).
+		Path(pathValidators).
+		ToJSON(&resp).
+		Fetch(ctx)
+
+	return resp, err
+}
+
+func (c *Client) AddValidators(ctx context.Context, shares ...ShareKeys) ([]web3signer.Status, error) {
+	encodedShares := make([]ShareKeys, 0, len(shares))
+	for _, share := range shares {
+		encodedShares = append(encodedShares, ShareKeys{
+			EncryptedPrivKey: share.EncryptedPrivKey,
+			PublicKey:        share.PublicKey,
+		})
+	}
+
+	req := AddValidatorRequest{
+		ShareKeys: encodedShares,
+	}
+
+	var resp web3signer.ImportKeystoreResponse
+	var errStr string
+	err := requests.
+		URL(c.baseURL).
+		Client(c.httpClient).
+		Path(pathValidators).
+		BodyJSON(req).
+		Post().
+		ToJSON(&resp).
+		AddValidator(requests.ValidatorHandler(requests.DefaultValidator, requests.ToString(&errStr))).
+		Fetch(ctx)
+
+	if requests.HasStatusErr(err, http.StatusUnprocessableEntity) {
+		return nil, ShareDecryptionError(errors.New(errStr))
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+
+	if len(resp.Data) != len(shares) {
+		return nil, fmt.Errorf("unexpected statuses length, got %d, expected %d", len(resp.Data), len(shares))
+	}
+
+	var statuses []web3signer.Status
+	for _, data := range resp.Data {
+		statuses = append(statuses, data.Status)
+	}
+
+	return statuses, nil
+}
+
+func (c *Client) RemoveValidators(ctx context.Context, sharePubKeys ...phase0.BLSPubKey) ([]web3signer.Status, error) {
+	req := web3signer.DeleteKeystoreRequest{
+		Pubkeys: sharePubKeys,
+	}
+
+	var resp web3signer.DeleteKeystoreResponse
+	err := requests.
+		URL(c.baseURL).
+		Client(c.httpClient).
+		Path(pathValidators).
+		BodyJSON(req).
+		Delete().
+		ToJSON(&resp).
+		Fetch(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+
+	if len(resp.Data) != len(sharePubKeys) {
+		return nil, fmt.Errorf("unexpected statuses length, got %d, expected %d", len(resp.Data), len(sharePubKeys))
+	}
+
+	var statuses []web3signer.Status
+	for _, data := range resp.Data {
+		statuses = append(statuses, data.Status)
+	}
+
+	return statuses, nil
+}
+
+func (c *Client) Sign(ctx context.Context, sharePubKey phase0.BLSPubKey, payload web3signer.SignRequest) (phase0.BLSSignature, error) {
+	var resp web3signer.SignResponse
+	err := requests.
+		URL(c.baseURL).
+		Client(c.httpClient).
+		Path(pathValidatorsSign + sharePubKey.String()).
+		BodyJSON(payload).
+		Post().
+		ToJSON(&resp).
+		Fetch(ctx)
+	if err != nil {
+		return phase0.BLSSignature{}, fmt.Errorf("request failed: %w", err)
+	}
+
+	return resp.Signature, nil
+}
+
```

Comment:
can't we have the same API as the ethereum specs and web3signer itself for this function?

Reply:
It's the same: https://ethereum.github.io/remote-signing-api/#/Signing/SIGN

If you see any difference, please let me know